api/v1alpha1/spaceprovisionerconfig_types.go

```diff
@@ -0,0 +1,92 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+const (
+	SpaceProvisionerConfigToolchainClusterNotFoundReason = "ToolchainClusterNotFound"
+	SpaceProvisionerConfigValidReason                    = "AllChecksPassed"
+)
+
+// +k8s:openapi-gen=true
+type SpaceProvisionerConfigSpec struct {
+	// PlacementRoles is the list of roles, or flavors, that the provisioner possesses that influence
+	// the space scheduling decisions.
+	// +optional
+	PlacementRoles []string `json:"placementRoles,omitempty"`
+
+	// ToolchainCluster is the name of the ToolchainCluster CR of the member cluster that this config is for.
+	ToolchainCluster string `json:"toolchainCluster"`
+
+	// Enabled specifies whether the member cluster is enabled (and therefore can hold spaces) or not.
+	// +optional
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled"`
+
+	// CapacityThresholds specifies the max capacities allowed in this provisioner
+	// +optional
+	CapacityThresholds SpaceProvisionerCapacityThresholds `json:"capacityThresholds"`
+}
+
+// SpaceProvisionerCapacityThresholds defines the capacity thresholds of the space provisioner
+// +k8s:openapi-gen=true
+type SpaceProvisionerCapacityThresholds struct {
+	// MaxNumberOfSpaces is the maximum number of spaces that can be provisioned to the referenced cluster.
+	//
+	// 0 or undefined value means no limit.
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	MaxNumberOfSpaces uint `json:"maxNumberOfSpaces,omitempty"`
+	// MaxMemoryUtilizationPercent is the maximum memory utilization of the cluster to permit provisioning
+	// new spaces to it.
+	//
+	// 0 or undefined value means no limit.
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=100
+	// +optional
+	MaxMemoryUtilizationPercent uint `json:"maxMemoryUtilizationPercent,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+type SpaceProvisionerConfigStatus struct {
+	// Conditions describes the state of the configuration (its validity).
+	// The only known condition type is "Ready".
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
btw, we have our own Condition type https://github.com/codeready-toolchain/api/blob/d5c7803671c1e0603b63761c07748d5910461962/api/v1alpha1/conditions.go#L28-L45 that we use in most of our CRDs, and we have a bunch of helper functions for that https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/condition/condition.go

I'm aware of this Condition from apimachinery. We can eventually replace most of the fields using the k8s conditon, but let's do it later as a separate task and let's stay what we have there for now.

Also, you can define the merge strategy (not sure what is the default one) and the merge key:
```suggestion
	// +optional
	// +patchMergeKey=type
	// +patchStrategy=merge
	// +listType=map
	// +listMapKey=type
	Conditions []Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```